decision/engine.go

```diff
@@ -408,13 +456,8 @@ func (e *Engine) MessageReceived(p peer.ID, m bsmsg.BitSwapMessage) {
 		} else {
 			log.Debugf("wants %s - %d", entry.Cid, entry.Priority)
 			l.Wants(entry.Cid, entry.Priority)
-			blockSize, err := e.bs.GetSize(entry.Cid)
-			if err != nil {
-				if err == bstore.ErrNotFound {
-					continue
-				}
-				log.Error(err)
-			} else {
+			blockSize := blockSizes[entry.Cid]
+			if blockSize > 0 {
```

Comment:
Can't we technically have zero-sized blocks? I mean, we shouldn't do that but I'd rather not _break_ it.

CID: bafkreihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku

Reply:
This should probably be:

```go
blockSize, have := blockSizes[entry.Cid]
```